pkg/solana/logpoller/types.go

```diff
@@ -0,0 +1,98 @@
+package logpoller
+
+import (
+	"database/sql/driver"
+	"encoding/json"
+	"fmt"
+
+	"github.com/gagliardetto/solana-go"
+)
+
+type PublicKey solana.PublicKey
+
+// Scan implements Scanner for database/sql.
+func (k *PublicKey) Scan(src interface{}) error {
+	return scanFixedLengthArray("PublicKey", solana.PublicKeyLength, src, k[:])
+}
+
+// Value implements valuer for database/sql.
+func (k PublicKey) Value() (driver.Value, error) {
+	return k[:], nil
+}
+
+func (k PublicKey) ToSolana() solana.PublicKey {
+	return solana.PublicKey(k)
+}
```

Comment:
I don't understand the need for the `ToSolana` methods. They don't follow a common interface since they all return something different so we don't really gain anything on using them. Casting, `solana.PublicKey(value)`, would do the same trick in every case given all of the structs are exported from the package.

Reply:
They are needed to ensure that users of the PublicKey type won't introduce bugs by casting to the wrong type.
The compiler won't show an error if user tries to cast `PublicKey` to  `solana.Hash`